### ISSUE TITLE
MCreator

### ISSUE THREAD

**Issue by jakobneufeld** [NixOS/nixpkgs#147479]
**Project description**
_describe the project a little_
Mcreator is a mod maker for minecraft. It is open-source. It is written in Java 11
**Metadata**

* homepage URL: https://mcreator.net/
* source URL:https://github.com/MCreator/MCreator
* license: mit, bsd, gpl2+ , gpl3 modified
* platforms: unix, linux, darwin, ... linux, darwin, windows

**Comment by MathiasSven** [comment]
Anyone looking for a bandage, I made an impure flake for you to install it using FHS
`nix profile install github:MathiasSven/MCreatorFlake --no-write-lock-file --impure`

**Comment by Krutonium** [comment]
Bump - I'd love to see this

**Comment by gro-david** [comment]
Same here, I really do need MCreator, for now I am stuck using it in a windows vm which is really not optimal :/

**Comment by dzoidbrg** [comment]
Maybe try distribox? I run all non nix compatible  software there. It’s super easy to use

**Comment by gro-david** [comment]
Since I only use MCretor for work related stuff and I also need Minecraft Education for that, which is completely Linux incompatible, I might as well only use one VM instead of juggling multiple, but thanks for the suggestion anyway.

**Comment by dzoidbrg** [comment]
Yeah in your case a VM makes sense of course. Just general advice I give to fellow Nixians :)

On February 7, 2025 8:01:01 PM GMT+01:00, "Dávid Groniewsky" ***@***.***> wrote:
>Since I only use MCreator for work related stuff and I also need Minecraft Education for that, which is completely Linux incompatible, I might as well only use one VM instead of juggling multiple, but thanks for the suggestion anyway. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/issues/147479#issuecomment-2643754361
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
Jakob Neufeld

**Comment by gro-david** [comment]
@dzoidbrg Did the flake that @MathiasSven provided work for you? For me it does not seem to even install, I opened an issue in his repo, but maybe you could help. This is the output:
```
warning: not writing modified lock file of flake 'github:MathiasSven/MCreatorFlake':
• Added input 'mcreator-releases':
    'https://api.github.com/repos/MCreator/MCreator/releases?narHash=sha256-jcM73K/kOXXP7ZKrIx6GZVRhccnB8LCIJV6NXsdvwMQ%3D'
• Added input 'nixpkgs':
    'github:NixOS/nixpkgs/a79cfe0ebd24952b580b1cf08cd906354996d547?narHash=sha256-mIvECo/NNdJJ/bXjNqIh8yeoSjVLAuDuTUzAo7dzs8Y%3D' (2025-02-08)
error:
       … in the left operand of the update (//) operator
         at /nix/store/igbfwr2gq5v3yplh8v48d0p77m7000cc-source/flake.nix:114:55:
          113|             mcreatorFromVersion (mostUptoDateFor base version).mcversion;
          114|         }) (allVersionsFromBase base)) baseVersions)) // {
             |                                                       ^
          115|           default = mcreatorFromVersion mostUptoDateForgeAny;

       … while evaluating a branch condition
         at /nix/store/j33wzkzndh41cyyy7i18bqm1srlv84cq-source/lib/attrsets.nix:1502:5:
         1501|     in
         1502|     if list == [ ] then
             |     ^
         1503|       # Calling binaryMerge as below would not satisfy its invariant

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: expected a list but found null: null
```

**Comment by dzoidbrg** [comment]
Oh I’ll try testing it out. I always run it through distrobox so I never tried.On Feb 10, 2025, at 07:06, Dávid Groniewsky ***@***.***> wrote:﻿
@dzoidbrg Did the flake that @MathiasSven provided work for you? For me it does not seem to even install, I opened an issue in his repo, but maybe you could help. This is the output:
warning: not writing modified lock file of flake 'github:MathiasSven/MCreatorFlake':
• Added input 'mcreator-releases':
    'https://api.github.com/repos/MCreator/MCreator/releases?narHash=sha256-jcM73K/kOXXP7ZKrIx6GZVRhccnB8LCIJV6NXsdvwMQ%3D'
• Added input 'nixpkgs':
    'github:NixOS/nixpkgs/a79cfe0ebd24952b580b1cf08cd906354996d547?narHash=sha256-mIvECo/NNdJJ/bXjNqIh8yeoSjVLAuDuTUzAo7dzs8Y%3D' (2025-02-08)
error:
       … in the left operand of the update (//) operator
         at /nix/store/igbfwr2gq5v3yplh8v48d0p77m7000cc-source/flake.nix:114:55:
          113|             mcreatorFromVersion (mostUptoDateFor base version).mcversion;
          114|         }) (allVersionsFromBase base)) baseVersions)) // {
             |                                                       ^
          115|           default = mcreatorFromVersion mostUptoDateForgeAny;

       … while evaluating a branch condition
         at /nix/store/j33wzkzndh41cyyy7i18bqm1srlv84cq-source/lib/attrsets.nix:1502:5:
         1501|     in
         1502|     if list == [ ] then
             |     ^
         1503|       # Calling binaryMerge as below would not satisfy its invariant

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: expected a list but found null: null

**Comment by AleXoundOS** [comment]
https://github.com/Cristiandis/mcreator-flake seems to work.

**Comment by dzoidbrg** [comment]
Yeah works for me too. I posted this 5 years ago; time flies lol
